wagtail/wagtailcore/models.py

```diff
@@ -537,15 +537,23 @@ def check(cls, **kwargs):
         from wagtail.wagtailadmin.forms import WagtailAdminPageForm
         if not issubclass(cls.base_form_class, WagtailAdminPageForm):
             errors.append(checks.Error(
-                "base_form_class does not extend WagtailAdminPageForm",
+                "{}.base_form_class does not extend WagtailAdminPageForm".format(
+                    cls.__name__),
                 hint="Ensure that {}.{} extends WagtailAdminPageForm".format(
                     cls.base_form_class.__module__,
                     cls.base_form_class.__name__),
                 obj=cls,
                 id='wagtailcore.E002'))
-        # Sadly, there is no way of checking the form class returned from
-        # cls.get_edit_handler().get_form_class(cls), as these calls can hit
-        # the DB in order to fetch content types.
+
+        edit_handler = cls.get_edit_handler()
+        if not issubclass(edit_handler.get_form_class(cls), WagtailAdminPageForm):
```

Comment:
Are you sure that we can perform this check? I have some doubts because of [this comment](https://github.com/torchbox/wagtail/pull/2405/files#diff-9ec0dccfdaa33a0fd310c76b5db2cb37L546).